lib/rubocop/cop/style/lambda.rb

```diff
@@ -38,6 +38,19 @@ def lambda_length(block_node)
 
           end_line - start_line
         end
+
+        def autocorrect(node)
+          expr = node.ancestors.first.loc.expression
+
+          @corrections << lambda do |corrector|
+            src = if node.loc.expression.source == 'lambda'
+                    expr.source.sub(/^lambda\s+\{\s*\|([^\|]+)\|/, '->(\\1) {')
```

Comment:
I was thinking that since you have to introduce special handling of the zero args case in `autocorrect` you might as well change the message too.